controller/linodemachine_controller_helpers.go

```diff
@@ -35,16 +37,61 @@ import (
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/handler"
+
+	infrav1alpha1 "github.com/linode/cluster-api-provider-linode/api/v1alpha1"
 )
 
+func (*LinodeMachineReconciler) newCreateConfig(ctx context.Context, machineScope *scope.MachineScope, tags []string, logger logr.Logger) (*linodego.InstanceCreateOptions, error) {
+	var err error
+
+	createConfig := linodeMachineSpecToInstanceCreateConfig(machineScope.LinodeMachine.Spec)
+	if createConfig == nil {
+		err = errors.New("failed to convert machine spec to create instance config")
+
+		logger.Error(err, "Panic! Struct of LinodeMachineSpec is different than InstanceCreateOptions")
+
+		return nil, err
+	}
+	createConfig.SwapSize = util.Pointer(0)
+
+	bootstrapData, err := machineScope.GetBootstrapData(ctx)
+	if err != nil {
+		logger.Info("Failed to get bootstrap data", "error", err.Error())
+
+		return nil, err
+	}
+	createConfig.Metadata = &linodego.InstanceMetadataOptions{
+		UserData: b64.StdEncoding.EncodeToString(bootstrapData),
+	}
+
+	if createConfig.Tags == nil {
+		createConfig.Tags = []string{}
+	}
+	createConfig.Tags = append(createConfig.Tags, tags...)
+
+	if createConfig.Label == "" {
+		createConfig.Label = util.RenderObjectLabel(machineScope.LinodeMachine.UID)
+	}
+
+	if createConfig.Image == "" {
+		createConfig.Image = reconciler.DefaultMachineControllerLinodeImage
+	}
+
+	if createConfig.RootPass == "" {
+		createConfig.RootPass = uuid.NewString()
```

Comment:
Using a UUID for the root password here gives me pause even if just a default if unset.

Reply:
is there a more secure alternative to this for generating a random password?

Reply:
@AshleyDumaine @eljohnson92 This change is temporary in the sens we have to design password management at all. Those passwords are not stored in anywhere (except Linode does what is does with it) and generated in the context of `immutable infrastructure`. On the other hand uuid uses `crypto/rand` which is cryptographically secure, so i'm open to other solutions, but please suggest a more secure random generation tool.

Reply:
@avestuk For me, even with `authorized_keys`, not setting `root_pass` for the LinodeMachine gives me the following:
```
status:
  conditions:
  - lastTransitionTime: "2024-02-07T17:45:25Z"
    message: '[400] [root_pass] root_pass is required'
    reason: CreateError
    severity: Error
    status: "False"
    type: Ready
  failureMessage: ""
  instanceState: offline
  ready: false
```